app/src/androidTest/java/org/mozilla/fenix/ui/SmokeTest.kt

```diff
@@ -62,4 +63,160 @@ class SmokeTest {
+        val youtubeUrl = "www.youtube.com"
+
+        navigationToolbar {
+        }.enterURLAndEnterToBrowser(defaultWebPage.url) {
+        }.openThreeDotMenu {
+            verifyThreeDotMainMenuItems()
+            verifySaveCollection()
+        }.clickAddOnsReportSiteIssue {
+            verifyUrl("webcompat.com/issues/new")
+        }.openTabDrawer {
+        }.openTab(defaultWebPage.title) {
+        }.openThreeDotMenu {
+        }.openHistory {
+            verifyTestPageUrl(defaultWebPage.url)
+        }.goBackToBrowser {
+        }.openThreeDotMenu {
+        }.openBookmarks {
+            verifyBookmarksMenuView()
+            verifyEmptyBookmarksList()
+        }.goBackToBrowser {
+        }.openThreeDotMenu {
+        }.openSyncedTabs {
+            verifyNavigationToolBarHeader()
+            verifySyncedTabsStatus()
+        }.goBack {
+        }.openThreeDotMenu {
+        }.openSettings {
+            verifySettingsView()
+        }.goBackToBrowser {
+        }.openThreeDotMenu {
+        }.openFindInPage {
+            verifyFindInPageSearchBarItems()
+        }.closeFindInPage {
+        }.openThreeDotMenu {
+        }.addToFirefoxHome {
+            verifySnackBarText("Added to top sites!")
+        }.openTabDrawer {
+        }.openHomeScreen {
+            verifyExistingTopSitesTabs(defaultWebPage.title)
+        }.openTabDrawer {
+        }.openTab(defaultWebPage.title) {
+        }.openThreeDotMenu {
+        }.openAddToHomeScreen {
+            verifyShortcutNameField(defaultWebPage.title)
+            clickAddShortcutButton()
+            clickAddAutomaticallyButton()
+        }.openHomeScreenShortcut(defaultWebPage.title) {
+        }.openThreeDotMenu {
+        }.openSaveToCollection {
+            verifyCollectionNameTextField()
+        }.goBackToBrowser {
+        }.openThreeDotMenu {
+        }.bookmarkPage {
+            verifySnackBarText("Bookmark saved!")
+        }.openThreeDotMenu {
+        }.sharePage {
+            verifyShareAppsLayout()
+        }.closeShareDialogReturnToPage {
+        }.openThreeDotMenu {
+        }.refreshPage {
+            verifyUrl(defaultWebPage.url.toString())
+        }.openTabDrawer {
+            closeTabViaXButton(defaultWebPage.title)
+        }.openHomeScreen {
+            navigationToolbar {
+            }.enterURLAndEnterToBrowser(youtubeUrl.toUri()) {
+                verifyBlueDot()
+            }.openThreeDotMenu {
+                verifyOpenInAppButton()
+            }
+        }
+    }
+
+    @Test
+    fun verifyPageMainMenuItemsListInPortraitPrivateModeTest() {
+        val defaultWebPage = TestAssetHelper.getGenericAsset(mockWebServer, 1)
+        // Add this to check openInApp and also youtube is a default app available in every Android emulator/device
+        val youtubeUrl = "www.youtube.com"
+
+        homeScreen {
+            togglePrivateBrowsingModeOnOff()
+            navigationToolbar {
+                navigationToolbar {

```

Comment:
Just realized about this...This navigationToolbar is duplicated, this or the above can be removed

Reply:
Yes have to remove that. It was a mistake over there